R/tm_g_swimlane.R

```diff
@@ -325,16 +326,16 @@ srv_g_swimlane <- function(id,
     output_q <- reactive({
       teal::validate_inputs(iv())
 
-      validate(need("ADSL" %in% names(data), "'ADSL' not included in data"))
+      validate(need("ADSL" %in% datanames(data()), "'ADSL' not included in data"))
       validate(need(
-        (length(data) == 1 && dataname == "ADSL") ||
-          (length(data) >= 2 && dataname != "ADSL"), paste(
+        (length(datanames(data())) == 1 && dataname == "ADSL") ||
+          (length(datanames(data())) >= 2 && dataname != "ADSL"), paste(
```

Comment:
Creating a `length` method for `teal_data` overkill?

Reply:
I am not sure if its used in too many places.

Reply:
Not overkill but wrong. What would the length represent? There are several features to a `teal_data` beyond `datanames`.

Reply:
Makes sense. Too ambiguous.